traversal/planner/Planner.java

```diff
@@ -28,19 +29,19 @@
 
 public interface Planner {
 
-    static Planner create(Structure structure) {
-        List<Structure> retrievedStructures = retrievedStructures(structure.asGraphs());
-        if (retrievedStructures.size() == 1) return ConnectedPlanner.create(retrievedStructures.get(0));
-        else return MultiPlanner.create(retrievedStructures);
+    static Planner create(Structure structure, Modifiers modifiers) {
+        List<Structure> retrievedStructures = splitStructure(structure, modifiers);
+        if (retrievedStructures.size() == 1) return ConnectedPlanner.create(retrievedStructures.get(0), modifiers);
+        else return MultiPlanner.create(retrievedStructures, modifiers);
     }
 
     /**
-     * Filter out structures that are not retrievable, such as disconnected type vertices
-     * TODO: We can further improve this if we have access to the filter as well
```

Comment:
we implement the optimisation that removes subgraphs which are not retrieved at all